pkg/storage/bucket/user_bucket_client.go

```diff
@@ -9,6 +9,13 @@ import (
 	"github.com/thanos-io/thanos/pkg/objstore"
 )
 
+// UserBucketOverrides defined config overrides supported by UserBucketClient.
+type UserBucketOverrides interface {
+	// S3SSEKMSKeyID returns the per-tenant S3 KMS-SSE key id or an empty string
+	// if no override is set.
+	S3SSEKMSKeyID(userID string) string
```

Comment:
Not sure how to improve this method name, but thank god for autocomplete. (Adding underscores would not work, right? Other than that, no good suggestion.)

Reply:
Yes, that name is ugly but don't think underscores are an option.

Reply:
One way might be to spell out SSE like `S3ServierSideEncryptionKMSKeyID` it's kinda long, but maybe it's more readable then bunch of `S` and `3` stuffed together :-) (thank got for autocomplete in this case too). 

Other ways may be to swap terms around like 

`SSEKMSKeyIDS3` but I don't think this helps much. 

Another approach may be to use nested interface like: 

```go
type S3ServerSideEncryptionConfigProvider interface {
     KMSKeyID() string
     KMSType() string
}

type TenantConfigProvider interface {
      S3ServerSideEncryptionConfig(user string) S3ServerSideEncryptionConfigProvider
}
```

This is a lot of typing, but maybe seeing `config.S3ServerSideEncryptionConfig(userId).KMSKeyID()` in the code is worth the typing?

Reply:
I've the feeling we're just going to add complexity for the sake of clarifying what SSE means, but if you're into Cortex internals I believe you quickly realise what it means.